cli/src/deno.rs

```diff
@@ -118,8 +154,23 @@ impl ModuleLoader for TypescriptModuleLoader {
                 _ => return Err(anyhow!("Unknown JS module format: {}", module_specifier)),
             };
 
-            // Read the source and transpile it if necessary.
-            let mut code = fs::read_to_string(&path)?;
+            // Load either a local file under the extensions directory, or a Deno standard library
+            // module. Reject all URLs that do not fit these two use cases.
+            let mut code = match (module_specifier.scheme(), module_specifier.host()) {
+                ("file", None) => {
+                    ExtensionsModuleLoader::load_from_filesystem(&module_specifier).await?
+                }
+                ("https", Some(Host::Domain(domain))) => {
+                    ExtensionsModuleLoader::load_from_deno_std(&module_specifier, domain).await?
```

Comment:
We're passing the `module_specifier` anyway. Why not handle the domain stuff internally?

Reply:
It would require another `match` statement and another couple of lines of error management in the `load_*` function. If we use the current solution instead, other invalid states are handled by the catch-all arm. Neither solution is perfectly clean looking, though.

Reply:
You already have a conditional anyway though? All you have to do is change your `== "deno.land` to a pattern match?

I don't think it's a big deal to have `None` and `!= deno.land` handled by the same error handler. As long as there's a generic "only deno.land is supported as domain" error message that applies perfectly well to any potential domain.

Reply:
That makes sense.